include/BigInt.hpp

```diff
@@ -93,6 +93,14 @@ class BigInt {
         // I/O stream operators:
         friend std::istream& operator>>(std::istream&, BigInt&);
         friend std::ostream& operator<<(std::ostream&, const BigInt&);
+        
+        
+        // gcd: 
+        friend BigInt gcd(const BigInt& num1, const BigInt& num2);
+        friend BigInt gcd(const BigInt& num1, const long long& num2);
+        friend BigInt gcd(const BigInt& num1, const std::string& num2);
```

Comment:
The GCD functions don't need to be friends of BigInt as they can be implemented without accessing the private members.